modules/ddb_cover_upload/includes/ddb_cover_upload.admin.inc

```diff
@@ -0,0 +1,663 @@
+  }
+
+  $form['ddb_cover_upload']['right']['results'] = array(
+    '#markup' => theme('ddb_cover_upload_search_preview', array('content' => $content)),
+  );
+
+  return $form;
+}
+
+/**
+ * Ajax form callback to render search preview.
+ *
+ * @param array $form
+ *   The form array.
+ * @param array $form_state
+ *   The state of the form.
+ *
+ * @return string
+ *   The rendered preview.
+ *
+ * @throws \Exception
+ */
+function ddb_cover_upload_search_preview(array $form, array $form_state) {
+  $id = $form_state['values']['ddb_cover_upload_material'];
+
+  $entity = ding_entity_load($id, 'ting_object');
+  $content = FALSE;
+  if ($entity) {
+    $content = ting_object_view($entity, 'teaser');
+  }
+
+  $form['ddb_cover_upload']['right']['results'] = array(
+    '#markup' => theme('ddb_cover_upload_search_preview', array('content' => $content)),
+  );
+
+  return $form['ddb_cover_upload']['right']['results'];
+}
+
+/**
+ * Image upload step of the form.
+ *
+ * @param array $form
+ *   The form array.
+ * @param array $form_state
+ *   The state of the form.
+ *
+ * @return array
+ *   The form for image upload.
+ *
+ * @throws \Exception
+ *   Theme exception error.
+ */
+function ddb_cover_upload_image_upload(array $form, array &$form_state) {
+  $path = drupal_get_path('module', 'ddb_cover_upload');
+
+  $form['#attached']['js'][] = array(
+    'data' => array(
+      'ddbCoverUpload' => array(
+        'bashPath' => $path,
+      ),
+    ),
+    'type' => 'setting',
+  );
+
+  // Attache Toast TUI image upload/editor.
+  $form['#attached']['css']['https://uicdn.toast.com/tui-color-picker/v2.2.3/tui-color-picker.css'] = array(
+    'type' => 'external',
+  );
+  $form['#attached']['css']['https://uicdn.toast.com/tui-image-editor/v3.8.0/tui-image-editor.css'] = array(
+    'type' => 'external',
+  );
+
+  $form['#attached']['js']['https://cdnjs.cloudflare.com/ajax/libs/fabric.js/3.3.2/fabric.js'] = array(
+    'type' => 'external',
+  );
+  $form['#attached']['js']['https://uicdn.toast.com/tui.code-snippet/v1.5.0/tui-code-snippet.min.js'] = array(
+    'type' => 'external',
+  );
+  $form['#attached']['js']['https://uicdn.toast.com/tui-color-picker/v2.2.3/tui-color-picker.js'] = array(
+    'type' => 'external',
+  );
+  $form['#attached']['js']['https://uicdn.toast.com/tui-image-editor/v3.8.0/tui-image-editor.js'] = array(
+    'type' => 'external',
+  );
+
+  // Custom JS to handle image upload and configuration of the image editor.
+  $form['#attached']['js'][$path . '/js/ddb_cover_upload_ui.admin.js'] = array(
+    'type' => 'file',
+  );
+
+  $form['image_editor'] = array(
+    '#markup' => theme('ddb_cover_upload_image_editor'),
+  );
+
+  $form['is'] = array(
+    '#type' => 'hidden',
+    '#value' => $form_state['step_information'][DDB_COVER_UPLOAD_STEP2]['stored_values']['ddb_cover_upload_material'],
+  );
+
+  $image = 'https://res.cloudinary.com/dandigbib/image/upload/v1576082092/default/forside-mangler-c.jpg';
```

Comment:
It will not change, as it is part of the CoverService default image (fallback image). But to be 110% sure I have download and added the image to the module :-).

Reply:
:+1: If not then I had expected the cover service to provide a more meaningful path.